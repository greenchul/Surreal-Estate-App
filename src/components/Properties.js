import React, { useEffect, useState } from "react";
import PropertyCard from "./PropertyCard";
import getProperties from "../requests/getProperties";
import "../styles/Properties.css";

const Properties = () => {
  const [listOfProperties, setListOfProperties] = useState([]);
  useEffect(() => {
    getProperties(setListOfProperties);
  }, []);
  return (
    <div className="properties">
      {listOfProperties.map((property) => {
        return (
          <div className="property">
            <PropertyCard
              key={property._id}
              title={property.title}
              type={property.type}
              bathrooms={parseInt(property.bathrooms, 10)}
              bedrooms={parseInt(property.bedrooms, 10)}
              price={parseInt(property.price, 10)}
              city={property.city}
              email={property.email}
            />
          </div>
        );
      })}
    </div>
  );
};

export default Properties;
