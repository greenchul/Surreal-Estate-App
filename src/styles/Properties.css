.properties {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
}

.property {
  width: 30%;
  border: 1px solid black;
  margin-bottom: 20px;
}
