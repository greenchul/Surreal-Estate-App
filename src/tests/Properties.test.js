import React from "react";
import { render } from "@testing-library/react";
import Properties from "../components/Properties";

describe("Properties Component", () => {
  test("component renders correctly", () => {
    const { asFragment } = render(<Properties />);

    expect(asFragment()).toMatchSnapshot();
  });
});
