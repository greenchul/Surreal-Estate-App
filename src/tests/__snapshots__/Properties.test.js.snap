// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties Component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="properties"
  />
</DocumentFragment>
`;
